shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/TemporaryTerminalRewriter.kt

```diff
@@ -0,0 +1,254 @@
+package com.mbta.tid.mbta_app.model
+
+import com.mbta.tid.mbta_app.model.response.GlobalResponse
+import com.mbta.tid.mbta_app.model.response.PredictionsStreamDataResponse
+import com.mbta.tid.mbta_app.model.response.ScheduleResponse
+
+/**
+ * For reasons, for mid-line subway shuttles/suspensions, schedules will correctly reflect the
+ * closure and use route patterns that stop at temporary terminals, but predictions will use the
+ * canonical route patterns. The way we work around this is that under the conditions specified in
+ * [appliesToRoute], we check the stops of the scheduled patterns against both the predicted pattern
+ * and the prediction's current stop to override the prediction's route pattern with a correct
+ * scheduled pattern, and then we load the headsign from the new route pattern's representative
+ * trip. This should be narrowly scoped enough to apply to temporary terminals but no other trips.
+ * Hopefully.
+ */
+class TemporaryTerminalRewriter(
+    val nearbyStaticData: NearbyStaticData,
+    val predictions: PredictionsStreamDataResponse,
+    val globalData: GlobalResponse,
+    val alerts: List<Alert>,
+    val schedules: ScheduleResponse
+) {
+    private val schedulesByRoute = schedules.schedules.groupBy { it.routeId }
+    // predictions that reflect auto-cancelled schedules will throw everything off, so filter out
+    // cancellations
+    val predictionsByRoute =
+        predictions.predictions.values
+            .filterNot { it.scheduleRelationship == Prediction.ScheduleRelationship.Cancelled }
+            .groupBy { it.routeId }
```

Comment:
question: why will these throw everything off? Don't we want to surface cancellations? It seems a bit risky to do this filtering outside of checking `appliesToRoute`